graphium/ipu/ipu_losses.py

```diff
@@ -1,12 +1,12 @@
 import torch
 from torch import Tensor
-from torch.nn import BCELoss, MSELoss, L1Loss
+from torch.nn import BCELoss, BCEWithLogitsLoss, MSELoss, L1Loss
 from torch._C import _infer_size
 from loguru import logger
 from graphium.trainer.losses import HybridCELoss
 
 
-class BCELossIPU(BCELoss):
+class BCELossIPU(BCEWithLogitsLoss):
```

Comment:
Change required for mixed-precision training.

Reply:
You have to be careful here. First, calling the class `BCELossWithLogitsIPU` for consistency. Then, changing the `spaces.py` file to include both `bce_with_logits` and `bce_with_logits_ipu`, and changing the YAML config accordingly. Also changing the output layer in the yaml `last_activation` parameter to `null`.